src/devices_models/device_constructors/hybrid_constructor.jl

```diff
@@ -0,0 +1,485 @@
+function construct_device!(
+    psi_container::OptimizationContainer,
+    sys::PSY.System,
+    model::DeviceModel{T, D},
+    ::Type{S},
+) where {
+    T <: PSY.HybridSystem,
+    D <: AbstractHybridFormulation,
+    S <: PM.AbstractActivePowerModel,
+}
+    devices = get_available_components(T, sys)
+
+    if !validate_available_devices(T, devices)
+        return
+    end
+
+    # Variables
+    add_variables!(psi_container, ActivePowerInVariable, devices)
```

Comment:
here you can implement a method add_variables that internally makes the variables for the hybrid systems components.